tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
-    return tools_ak
+    assert pre_upgrade_repo.sync()['result'] == 'success'
+    return pre_upgrade_repo
 
 
-def create_yum_test_repo(product_name, repo_url, org):
-    """Creates yum repo from given url and syncs
+@pytest.fixture(scope='module')
+def post_upgrade_repo(default_org):
+    """Enable custom errata repository"""
+    post_upgrade_repo = entities.Repository(
+        url=FAKE_9_YUM_REPO, product=entities.Product(organization=default_org).create()
+    ).create()
+    assert post_upgrade_repo.sync()['result'] == 'success'
+    return post_upgrade_repo
 
-    :param str product_name: Product name to be created
-    :param str repo_url: The repo url for repo
-    :param nailgun.entities.Organization org: Organization
 
-    :return tuple(nailgun entities): Returns product and yum_repo
-    """
-    product = entities.Product(name=product_name, organization=org).create()
-    yum_repo = entities.Repository(
-        name=f'{product_name}_repo', product=product, url=repo_url, content_type='yum'
+@pytest.fixture(scope='module')
+def pre_upgrade_cv(default_org, pre_upgrade_repo):
+    """Create and publish repo"""
+    pre_upgrade_cv = entities.ContentView(
+        organization=default_org, repository=[pre_upgrade_repo.id]
     ).create()
-    yum_repo.sync()
-    return product, yum_repo
+    pre_upgrade_cv.publish()
+    pre_upgrade_cv = pre_upgrade_cv.read()
+    return pre_upgrade_cv
 
 
-def update_product_subscription_in_ak(product, yum_repo, ak, org):
-    """Updates given products subscription in given AK
-
-    :param nailgun.entities.Product product: products name to calculate
-        subscription id
-    :param nailgun.entities.Repository yum_repo: yum repository
-    :param nailgun.entities.ActivationKey ak: Ak
-    :param nailgun.entities.Organization org: Organization
-    """
-    cv_from_ak = ak.content_view
-    cv = cv_from_ak.read()
-    cv.repository.append(yum_repo)
-    cv = cv.update(['repository'])
-    cv.publish()
-    cv = cv.read()  # Published CV with new version
-    # Promote CV
-    environment = (
-        entities.ActivationKey(organization=org)
-        .search(query={'search': f'name={ak.name}'})[0]
-        .environment
+@pytest.fixture(scope='module')
+def post_upgrade_cv(default_org, post_upgrade_repo):
+    """Create and publish repo"""
+    post_upgrade_cv = entities.ContentView(
+        organization=default_org, repository=[post_upgrade_repo.id]
+    ).create()
+    post_upgrade_cv.publish()
+    post_upgrade_cv = post_upgrade_cv.read()
+    return post_upgrade_cv
+
+
+@pytest.fixture(scope='module')
+def module_ak(default_org, module_lce_library, pre_upgrade_repo):
+    """Create a module AK in Library LCE"""
+    ak = entities.ActivationKey(
+        content_view=DEFAULT_CV,
+        environment=module_lce_library,
+        organization=default_org,
+    ).create()
+    # Fetch available subscriptions
+    subs = entities.Subscription(organization=default_org).search()
```

Comment:
Same for this `default_subscription` fixture ?

Reply:
I don't consider this necessary for the PR scope.